src/Network/Network/Generated/LoadBalancer/LoadBalancerRule/NewAzureRmLoadBalancerRuleConfigCommand.cs

```diff
@@ -179,16 +185,39 @@ public override void Execute()
                 }
                 vLoadBalancingRules.FrontendIPConfiguration.Id = this.FrontendIpConfigurationId;
             }
-            if(!string.IsNullOrEmpty(this.BackendAddressPoolId))
+
+            if (this.BackendAddressPoolId.Length > 0)
             {
-                // BackendAddressPool
-                if (vLoadBalancingRules.BackendAddressPool == null)
+                if (this.BackendAddressPoolId.Length == 1)
+                {
+                    // BackendAddressPool
+                    if (vLoadBalancingRules.BackendAddressPool == null)
+                    {
+                        vLoadBalancingRules.BackendAddressPool = new PSResourceId();
+                    }
+
+                    vLoadBalancingRules.BackendAddressPool.Id = this.BackendAddressPoolId[0];
+                }
+                else if (this.BackendAddressPoolId.Length == 2)
+                {
+                    // BackendAddressPools
+                    if (vLoadBalancingRules.BackendAddressPools == null)
+                    {
+                        vLoadBalancingRules.BackendAddressPools = new List<PSResourceId>();
+                    }
+
+                    foreach (string backendAddressPoolId in this.BackendAddressPoolId)
+                    {
+                        vLoadBalancingRules.BackendAddressPools.Add(new PSResourceId() { Id = backendAddressPoolId });
+                    }
+                }
+                else
                 {
-                    vLoadBalancingRules.BackendAddressPool = new PSResourceId();
+                    throw new ArgumentException("BackendAddressPool cannot be more than 2");
```

Comment:
Why BackendAddressPool cannot be more than 2?
Besides, please throw AzPSArgumentException, see https://github.com/Azure/azure-powershell-common/tree/master/src/Common/Exceptions

Reply:
out backend cannot take more than 2.
this feature will allow 2 instead of one